faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -330,23 +336,21 @@ private Value handleResponse(FullHttpResponse response) {
       return json.treeToValue(resource, Value.class);
     } catch (IOException ex) {
       throw new AssertionError(ex);
-    } finally {
-      response.release();
     }
   }
 
   private CompletableFuture<Value> performRequest(JsonNode body, Optional<Duration> queryTimeout) {
     try {
         return handleNetworkExceptions(connection.post("", body, queryTimeout).thenApply(this::handleResponse));
-    } catch (IOException ex) {
+    } catch (Exception ex) {
         CompletableFuture<Value> oops = new CompletableFuture<>();
         oops.completeExceptionally(ex);
         return oops;
     }
   }
 
-  private void handleQueryErrors(FullHttpResponse response) {
-    int status = response.status().code();
+  private void handleQueryErrors(HttpResponse<String> response) {
+    int status = response.statusCode();
     if (status >= 300) {
       try {
```

Comment:
The error handling in this method can be significantly cleaned up: parseResponseBody seems like it could be rewritten to not throw and instead return null or some other failure sentinel if parsing fails. This would allow removing the very unfortunate catch on what is now line 383.